substrate/client/network/sync/src/lib.rs

```diff
@@ -1107,11 +1099,8 @@ where
 		});
 
 		let blocks = self.ready_blocks();
-		if let Some(OnBlockData::Import(origin, blocks)) =
-			(!blocks.is_empty()).then(|| self.validate_and_queue_blocks(blocks, false))
-		{
-			self.import_blocks(origin, blocks);
-		}
+
+		(!blocks.is_empty()).then(|| self.validate_and_queue_blocks(blocks, false))
```

Comment:
`let blocks = self.ready_blocks();` is kind of useless now and you could directly use it instead of creating `blocks`.

Reply:
I kept it because there is a check `!blocks.is_empty()`, and there is no `map()` for `Vec`. Am I missing something?

Reply:
What I meant is that why can't the code be:

```rust
(!self.blocks.is_empty()).then(|| self.validate_and_queue_blocks(blocks, false))
```

Reply:
I don't understand. We need to both check that `ready_blocks()` are not empty and also pass `ready_blocks()` to `validate_and_queue_blocks()`. How can we do that without an intermediate variable?

Reply:
Sorry I was being a dumbass, I didn't notice the parameter. Nevermind